src/AggregateFunctions/AggregateFunctionGroupSortedArray.h

```diff
@@ -0,0 +1,182 @@
+#pragma once
+
+#include <Columns/ColumnArray.h>
+#include <DataTypes/DataTypeArray.h>
+
+#include <AggregateFunctions/IAggregateFunction.h>
+#include "AggregateFunctionGroupSortedArrayData.h"
+
+namespace DB
+{
+template <typename TT, bool is_plain_column>
+inline TT readItem(const IColumn * column, Arena * arena, size_t row)
+{
+    if constexpr (std::is_same_v<TT, StringRef>)
+    {
+        if constexpr (is_plain_column)
+        {
+            StringRef str = column->getDataAt(row);
+            auto ptr = arena->alloc(str.size);
+            std::copy(str.data, str.data + str.size, ptr);
+            return StringRef(ptr, str.size);
+        }
+        else
+        {
+            const char * begin = nullptr;
+            return column->serializeValueIntoArena(row, *arena, begin);
+        }
+    }
+    else
+    {
+        return column->getUInt(row);
+    }
+}
+
+template <bool is_plain_column, typename T, bool is_weighted>
+class AggregateFunctionGroupSortedArray : public IAggregateFunctionDataHelper<
+                                              AggregateFunctionGroupSortedArrayData<T, is_weighted>,
+                                              AggregateFunctionGroupSortedArray<is_plain_column, T, is_weighted>>
+{
+protected:
+    using State = AggregateFunctionGroupSortedArrayData<T, is_weighted>;
+
+    UInt64 threshold;
+    DataTypePtr & input_data_type;
+    mutable std::mutex mutex;
+
+    static void deserializeAndInsert(StringRef str, IColumn & data_to);
+
+public:
+    AggregateFunctionGroupSortedArray(UInt64 threshold_, UInt64 /*load_factor*/, const DataTypes & argument_types_, const Array & params)
+        : IAggregateFunctionDataHelper<AggregateFunctionGroupSortedArrayData<T, is_weighted>, AggregateFunctionGroupSortedArray>(
+            argument_types_, params)
+        , threshold(threshold_)
+        , input_data_type(this->argument_types[0])
+    {
+    }
+
+    void create(AggregateDataPtr place) const override
+    {
+        IAggregateFunctionDataHelper<AggregateFunctionGroupSortedArrayData<T, is_weighted>, AggregateFunctionGroupSortedArray>::create(
+            place);
+        this->data(place).threshold = threshold;
+    }
+
+    String getName() const override { return is_weighted ? "groupSortedArrayWeighted" : "groupSortedArray"; }
```

Comment:
constexpr

Reply:
I understand this is a template and the compilator will translate to **return (TRUE):'A':'B'** so my guess it will be compiled as **return 'A'**.
Even more, as far as I know using constexpt with ternary operator is not allowed, so I have to use if constexpr...

Reply:
constexpr avoids the need to guess what the compiler will do. You don't need a ternary operator here at all